src/dfx/src/lib/builders/motoko.rs

```diff
@@ -132,6 +138,14 @@ impl CanisterBuilder for MotokoBuilder {
         std::fs::create_dir_all(idl_dir_path)
             .with_context(|| format!("Failed to create {}.", idl_dir_path.to_string_lossy()))?;
 
+        // If the management canister is being imported, emit the candid file.
+        if get_imports(cache.as_ref(), input_path)?
```

Comment:
Will this emit the management canister candid file if the import comes not from the "main" motoko file, but from a .mo file that it imports?

The best way to make sure would be to add an e2e test.

Reply:
Apologies, getting back to this now. Will address this issue.